### DISCUSSION TITLE
Persistent datastore

### DISCUSSION THREAD

**Discussion by a4smanjorg5** [libp2p/js-libp2p#2610] [Q&A]
How to use level-db via react native (android and iOS) to libp2p for persistent datastore? Is there a third-party module?

**Comment by achingbrain** [comment]
You can use [datastore-level](https://www.npmjs.com/package/datastore-level) to use level as a persistent datastore.  You can pass an instance of it as the `datastore` key of the options object to `createLibp2p`.

This would be really useful to add to the [react-native demo](https://github.com/ipfs-shipyard/js-libp2p-react-native) - could you post a link to some code when you get it running or open a PR adding it to the demo repo?

**Comment by abuvanth** [comment]
leveldb depends on indexdb but indexdb not available in react native.
we are using react-native-leveldb-level-adapter as leveldb alternative.

checkout https://github.com/cyberfly-io/cyberfly-orbitdb-react-native/blob/main/src/blockstore-level.ts
